### PR TITLE
Янчевский Сергей Б12-513

### PR DESCRIPTION
**Pull request by Ssssserge**
Решаю задачки...

### CODE REVIEW

File: task_01/build/task_01_tests[1]_include.cmake [hunk 1]
```diff
@@ -0,0 +1,5 @@
+if(EXISTS "/workspaces/spring_alogorithms/task_01/build/task_01_tests[1]_tests.cmake")
```
Comment (LostPointer) [inline task_01/build/task_01_tests[1]_include.cmake:1]: 🔴 **Build artifact закоммичен**

Файл содержит хардкоженный абсолютный путь `/workspaces/spring_alogorithms/` — внутренний артефакт devcontainer, ломает сборку в любом другом окружении. Папка `build/` должна быть в `.gitignore`. Удалить файл из PR.

File: task_02/tests/early_transition.in [hunk 2]
```diff
@@ -1,2 +1,2 @@
-5
-0 1 1 1 1
+5 1 1 1 1
```
Comment (LostPointer) [inline task_02/tests/early_transition.in:1]: 🔴 **Формат тестового файла сломан**

`main.cpp` читает `n` отдельно, затем массив. После изменения `n = 5`, а массив читается как `{1, 1, 1, 1, 0}` — нулей перед единицами нет, невалидный ввод.

Нужно вернуть исходный формат:
```
5
0 1 1 1 1
```

File: task_01/src/FindSum.hpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#include <vector>
```
Comment (LostPointer) [inline task_01/src/FindSum.hpp:1]: 🔴 **Отсутствует `#pragma once`** — без header guard двойное включение даёт дублирующиеся определения → ошибка компиляции.

File: task_01/src/FindSum.hpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+std::pair<int, int> find_pair_with_sum(int target_sum, std::vector<int> arr) {
```
Comment (LostPointer) [inline task_01/src/FindSum.hpp:3]: 🟠 **snake_case и вектор по значению**

Два замечания:
1. `find_pair_with_sum` — snake_case нарушает Google style. Нужно PascalCase: `FindPairWithSum`.
2. `std::vector<int> arr` — лишнее копирование вектора при каждом вызове. Нужно:
```cpp
std::pair<int,int> FindPairWithSum(int target_sum, const std::vector<int>& arr)
```

File: task_02/src/FindIndex.hpp [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#include <vector>
```
Comment (LostPointer) [inline task_02/src/FindIndex.hpp:1]: 🔴 **Отсутствует `#pragma once`**

File: task_02/src/FindIndex.hpp [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+
+
+int FindIndex(std::vector<int> vec){
```
Comment (LostPointer) [inline task_02/src/FindIndex.hpp:5]: 🟠 **Вектор по значению** — функция не модифицирует `vec`, нужно `const std::vector<int>& vec`.

File: task_01/src/main.cpp [hunk 7]
```diff
@@ -1,3 +1,32 @@
 #include <iostream>
 
-int main() { return 0; }
+#include "FindSum.hpp"
+
+
+#include <vector>
+
+using namespace std;
```
Comment (LostPointer) [inline task_01/src/main.cpp:8]: 🟠 **`using namespace std;`** — запрещён Google style. Использовать явные префиксы `std::`.

File: task_01/src/test.cpp [hunk 8]
```diff
@@ -1,5 +1,30 @@
 #include <gtest/gtest.h>
+#include <vector>
+#include "FindSum.hpp"
 
-TEST(Test, Simple) {
-  ASSERT_EQ(1, 1);  // Stack []
-}
\ No newline at end of file
+
+TEST(TwoPointersTest, BasicCase) {
```
Comment (LostPointer) [inline task_01/src/test.cpp:6]: 🟠 **Только 3 теста, нет edge cases**

Добавить как минимум:
- Пустой массив → `{-1, -1}`
- Массив из 1 элемента → `{-1, -1}`
- Два элемента, сумма совпадает

**Review by LostPointer — CHANGES_REQUESTED** [review]
## Обзор PR

Алгоритм `FindIndex` (task_02) корректен. Но есть критические проблемы: сломан тестовый файл и закоммичен build-артефакт.

**Хорошее:**
- `FindIndex`: бинарный поиск корректен, граничный случай `right == left + 1` обработан
- Тест на отрицательные числа в task_01 есть

**Форматирование:** 4-пробельные отступы, лишние пустые строки, нет trailing newline. Запустить `./format.sh`.
